src/main/java/org/variantsync/diffdetective/variationtree/VariationTree.java

```diff
@@ -0,0 +1,121 @@
+        this.root = root;
+        this.source = source;
+
+        Assert.assertTrue(root.isRoot());
+    }
+
+    /**
+     * Same as {@link fromFile(Path, boolean, CPPAnnotationParser)} but with
+     * a {@link CPPAnnotationParser#Default default annotation parser}.
+     */
+    public static VariationTree fromFile(
+        final Path p,
+        final boolean collapseMultipleCodeLines
+    ) throws IOException, DiffParseException {
+        return fromFile(p, collapseMultipleCodeLines, CPPAnnotationParser.Default);
+    }
+
+    /**
+     * Same as {@link fromFile(BufferedReader, VariationTreeSource, boolean, CPPAnnotationParser)}
+     * but registers {@code path} as source.
+     */
+    public static VariationTree fromFile(
+        final Path path,
+        final boolean collapseMultipleCodeLines,
+        final CPPAnnotationParser annotationParser
+    ) throws IOException, DiffParseException {
+        try (BufferedReader file = Files.newBufferedReader(path)) {
+            return fromFile(
+                file,
+                new LocalFileSource(path),
+                collapseMultipleCodeLines,
+                annotationParser
+            );
+        }
+    }
+
+    /**
+     * Same as {@link fromFile(BufferedReader, VariationTreeSource, boolean, CPPAnnotationParser)}
+     * but with a {@link CPPAnnotationParser#Default default annotation parser}.
+     */
+    public static VariationTree fromFile(
+        final BufferedReader input,
+        final VariationTreeSource source,
+        final boolean collapseMultipleCodeLines
+    ) throws IOException, DiffParseException {
+        return fromFile(
+            input,
+            source,
+            collapseMultipleCodeLines,
+            CPPAnnotationParser.Default
+        );
+    }
+
+    /**
+     * Parses a {@code VariationTree} from source code with C preprocessor annotations.
+     *
+     * @param input the source code to be parsed
+     * @param collapseMultipleCodeLines Set to true if subsequent lines of source code with the same
+     * {@link NodeType type} should be collapsed into a single artifact node representing all lines
+     * at once.
+     * @param annotationParser the parser that is used to parse preprocessor expressions
+     * @return a new {@code VariationTree} representing {@code input}
+     * @throws IOException if {@code input} throws {@code IOException}
+     * @throws DiffParseException if some preprocessor annotations can't be parsed
+     */
+    public static VariationTree fromFile(
+        final BufferedReader input,
+        final VariationTreeSource source,
+        final boolean collapseMultipleCodeLines,
+        final CPPAnnotationParser annotationParser
+    ) throws IOException, DiffParseException {
+        VariationTreeNode tree = DiffTreeParser
+            .createVariationTree(input, collapseMultipleCodeLines, false, annotationParser)
+            .getRoot()
+            .projection(BEFORE)
```

Comment:
Leave a note here that BEFORE and AFTER yield equivalent results and so just any of them is fine?